src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -63,12 +68,112 @@ class PlannerPlanAddCommand extends GraphCommand {
         cb();
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
   }
 
+  private updatePlanDetails(options: Options, newPlan: PlannerPlan): Promise<PlannerPlan & PlannerPlanDetails> {
+    const planId = newPlan.id as string;
+
+    if (!options.shareWithUserIds && !options.shareWithUserNames) {
+      return Promise.resolve(newPlan);
+    }
+
+    return Promise
+      .all([this.generateSharedWith(options), this.getPlanDetailsEtag(planId)])
+      .then(resArray => {
+        const sharedWith = resArray[0];
+        const etag = resArray[1];
+        const requestOptionsPlanDetails: any = {
+          url: `${this.resource}/v1.0/planner/plans/${planId}/details`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none',
+            'If-Match': etag,
+            'Prefer': 'return=representation'
+          },
+          responseType: 'json',
+          data: {
+            sharedWith: sharedWith
+          }
+        };
+
+        return request.patch(requestOptionsPlanDetails);
+      })
+      .then(planDetails => {
+        return { ...newPlan, ...planDetails as PlannerPlanDetails };
+      });
+  }
+
+  private getPlanDetailsEtag(planId: string): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans/${planId}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => response['@odata.etag']);
+  }
+
+  private generateSharedWith(options: Options): Promise<{ [userId: string]: boolean }> {
+    const sharedWith: { [userId: string]: boolean } = {};
+
+    return this
+      .getUserIds(options)
+      .then((userIds) => {
+        userIds.map(x => sharedWith[x] = true);
+
+        return Promise.resolve(sharedWith);
+      });
+  }
+
+  private getUserIds(options: Options): Promise<string[]> {
+    if (options.shareWithUserIds) {
+      return Promise.resolve(options.shareWithUserIds.split(','));
+    }
+
+    // Hitting this section means assignedToUserNames won't be undefined
+    const userNames = options.shareWithUserNames as string;
+    const userArr: string[] = userNames.split(',').map(o => o.trim());
+    let userIds: string[] = [];
+
+    const promises: Promise<{ value: User[] }>[] = userArr.map(user => {
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/users?$filter=userPrincipalName eq '${formatting.encodeQueryParameter(user)}'&$select=id,userPrincipalName`,
+        headers: {
+          'content-type': 'application/json'
+        },
+        responseType: 'json'
+      };
+
+      return request.get(requestOptions);
+    });
+
+    return Promise
+      .all(promises)
+      .then((usersRes: { value: User[] }[]): Promise<string[]> => {
+        let userUpns: string[] = [];
+
+        userUpns = usersRes.map(res => res.value[0]?.userPrincipalName as string);
```

Comment:
Instead of defining `userUpns` above as a variable with an empty array, we can move its definition here and change it to const.